src/mock_trie.c

```diff
@@ -0,0 +1,105 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+#include "mock_trie.h"
+#include "utils.h"
+
+/*
+* See mock_trie.h
+*/
+trie_t* trie_new(){
+    trie_t *t;
+    t = malloc(sizeof(trie_t));
+    int rc;
+
+    if (t == NULL) {
+      error("Could not allocate memory");
+      return NULL;
+    }
+
+    rc = trie_init(t);
+    if(rc != 1)
```

Comment:
Should be `EXIT_SUCCESS`/`EXIT_FAILURE` as appropriate

Reply:
The use of 1 and 0 here is inconsistent with `dict_new()`. This must be consistent.